base/linalg/blas.jl

```diff
@@ -339,6 +369,15 @@ nrm2(x::Array) = nrm2(length(x), pointer(x), 1)
     asum(n, X, incx)
 
 Sum of the absolute values of the first `n` elements of array `X` with stride `incx`.
+
+# Example:
+```jldoctest
+julia> Base.BLAS.asum(10, im*ones(10), 2)
+5.0
```

Comment:
It's hard to notice since I think we frequently get lucky that the space past the end of the array is happening to be set to zero. In C you might see nondeterministic results more often. Or you could be overly clever and take advantage of this intentionally with some subarray schemes.

I think the conclusion here is that BLAS functions don't do a whole lot of bounds checking if you haven't passed the array length. This potential issue may or may not be worth mentioning in the Julia docs, dunno. We should change any examples we give so that we're only reading from within the bounds of the arrays we create.